contracts/tips/src/contract.rs

```diff
@@ -0,0 +1,2044 @@
+    use desmos_bindings::posts::query::PostsQuery;
+    use desmos_bindings::query::DesmosQuery;
+    use desmos_bindings::subspaces::mocks::mock_subspaces_query_response;
+    use desmos_bindings::subspaces::query::SubspacesQuery;
+    use std::marker::PhantomData;
+
+    const ADMIN: &str = "admin";
+    const USER_1: &str = "user1";
+    const USER_2: &str = "user2";
+    const USER_3: &str = "user3";
+
+    fn init_contract(
+        deps: DepsMut<DesmosQuery>,
+        subspace_id: u64,
+        service_fee: Option<ServiceFee>,
+        saved_tips_threshold: u32,
+    ) -> Result<Response<DesmosMsg>, ContractError> {
+        instantiate(
+            deps,
+            mock_env(),
+            mock_info(ADMIN, &[]),
+            InstantiateMsg {
+                admin: ADMIN.to_string(),
+                subspace_id: subspace_id.into(),
+                service_fee,
+                saved_tips_threshold,
+            },
+        )
+    }
+
+    fn tip_user(
+        deps: DepsMut<DesmosQuery>,
+        from: &str,
+        to: &str,
+        founds: &[Coin],
+        coins: &[Coin],
+    ) -> Result<Response<DesmosMsg>, ContractError> {
+        let info = mock_info(from, founds);
+        execute(
+            deps,
+            mock_env(),
+            info,
+            ExecuteMsg::SendTip {
+                target: Target::UserTarget {
+                    receiver: to.to_string(),
+                },
+                amount: coins.to_vec(),
+            },
+        )
+    }
+
+    fn tip_post(
+        deps: DepsMut<DesmosQuery>,
+        from: &str,
+        post_id: u64,
+        founds: &[Coin],
+        coins: &[Coin],
+    ) -> Result<Response<DesmosMsg>, ContractError> {
+        let info = mock_info(from, founds);
+        execute(
+            deps,
+            mock_env(),
+            info,
+            ExecuteMsg::SendTip {
+                target: Target::ContentTarget {
+                    post_id: post_id.into(),
+                },
+                amount: coins.to_vec(),
+            },
+        )
+    }
+
+    fn get_tips_record_items(deps: DepsMut<DesmosQuery>) -> Vec<StateTip> {
+        tips_record()
+            .range(deps.storage, None, None, Order::Ascending)
+            .collect::<StdResult<Vec<(String, StateTip)>>>()
+            .unwrap()
+            .drain(0..)
+            .map(|(_, tip)| tip)
+            .collect()
+    }
+
+    #[test]
+    fn init_contract_with_invalid_subspace_id() {
+        let mut deps = mock_dependencies_with_custom_querier(&[]);
+
+        let init_err = init_contract(
+            deps.as_mut(),
+            0,
+            Some(ServiceFee::Fixed { amount: vec![] }),
+            1,
+        )
+        .unwrap_err();
+        assert_eq!(ContractError::InvalidSubspaceId {}, init_err);
+    }
+
+    #[test]
+    fn init_contract_with_non_existing_subspace() {
+        let querier = MockQuerier::<DesmosQuery>::new(&[(MOCK_CONTRACT_ADDR, &[])])
```

Comment:
Great method to mock deps. I think we can have this method in desmos-bindings allows devs to mock what they need instead of the default.

Reply:
Yes! was thikning the same, we can have something like `MockDesmosQuerier::new(...)` and with methods to mock the response of each module like `with_custom_subspaces_handler` or `with_custom_profiles_handler`, what you think?